### ISSUE TITLE
[User Teams] Teams entity

### ISSUE THREAD

**Issue by kwa20** [hzi-braunschweig/SORMAS-Project#6623]

**Comment by bernardsilenou** [comment]
@kwa20 
This is a nice one and very similar to event group module. If there is no specific reason, then I would suggest we call it user group.
Question please:
- In a group ( team), should all the users have the same right or role ? because this has an implication. From what you explained in another issue, then benefit of grouping user may be be that we want to send a task to the group for example, thus if all users in a groups does not have the same basic right, then some members may not be able to do certain task. 
- A first suggestion is if we can add the option that when creating a group, the user has to set the user roles/ right of the group,  and only users having all the rights of the group can be added to the group. If a user group does not have a right to do a certain task (eg case classification), then system should prevent users sending such task to the group.
- In summary, add a group user rignt == sum of all the rights of the user roles of the group, and when assigning task or users to a group, check if the rights are OK with that of the group.

**Comment by kwa20** [comment]
@bernardsilenou I'll respond to your comment here https://github.com/hzi-braunschweig/SORMAS-Project/issues/6622 so we can keep discussions in the concept issue and update linked issues if needed. This will be easier for developers later on.
